components/SemesterList.tsx

```diff
@@ -0,0 +1,257 @@
+    expandedArr.push(setExpanded);
+    const [props, drop] = useDrop({
+      accept: [ItemTypes.COURSE],
+      drop: () => { },
+      collect: (monitor) => {
+        if (monitor.isOver({ shallow: true })) {
+          expandedArr.forEach((v, index) => v(index === i));
+        }
+      }
+    });
+    parts.push(
+      <Accordion expanded={expanded} onChange={(_, e) => setExpanded(e)}>
+        <div ref={drop}>
+          <AccordionSummary sx={{ bgcolor: "primary.main" }}>
+            Year {i}
+          </AccordionSummary>
+        </div>
+        <AccordionDetails sx={{ p: 0 }}>
+          <div
+            style={{
+              display: "grid",
+              gridTemplateColumns: "repeat(4, 1fr)",
+              gridTemplateRows: "100%"
+            }}
+          >
+            {semesters
+              .filter((sem) => sem.year === i + 1)
+              .sort(sortSemester)
+              .map((sem, index) => (
+                <Semester
+                  key={index}
+                  accept={sem.accepts}
+                  onDrop={(item) => handleDrop(sem.semesterNumber, item)}
+                  semesterNumber={sem.semesterNumber}
+                  courses={sem.courses}
+                  SemesterCredits={sem.SemesterCredits}
+                  Warning={sem.Warning}
+                  warningPrerequisiteCourses={warningPrerequisiteCourses}
+                  warningFallvsSpringCourses={warningFallvsSpringCourses}
+                  warningDuplicateCourses={warningDuplicateCourses}
+                  year={sem.year}
+                  season={sem.season}
+                />
+              ))}
+          </div>
+        </AccordionDetails>
+      </Accordion>
+    );
+  }
+
+  return (
+    <div
+      className="generic"
+      style={{ overflow: "scroll auto", height: "100%" }}
+    >
+      {parts}
+    </div>
+    // <>
+    //   <div style={{ overflow: "hidden", clear: "both" }}>
+    //     {semesters.map(
+    //       (
+    //         {
+    //           accepts,
+    //           semesterNumber,
+    //           courses,
+    //           SemesterCredits,
+    //           Warning,
+    //           year,
+    //           season
+    //         },
+    //         index
+    //       ) => (
+    //         <Semester
+    //           accept={accepts}
+    //           onDrop={(item) => handleDrop(index, item)}
+    //           semesterNumber={semesterNumber}
+    //           courses={courses}
+    //           key={index}
+    //           SemesterCredits={SemesterCredits}
+    //           Warning={Warning}
+    //           warningPrerequisiteCourses={warningPrerequisiteCourses}
+    //           warningFallvsSpringCourses={warningFallvsSpringCourses}
+    //           warningDuplicateCourses={warningDuplicateCourses}
+    //           year={year}
+    //           season={season}
+    //         />
+    //       )
+    //     )}
+    //   </div>
+    // </>
+  );
+}
```

Comment:
1.  Can we get some comments in this file?
2. I see that you are still calculating the semester credit warning, but are no longer using it? What's up with that?
3. What's with the big comment in the return?